proto/cork/v2/proposal.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+package cork.v2;
+
+import "cosmos_proto/cosmos.proto";
+import "cork/v2/cork.proto";
+
+option go_package = "github.com/peggyjv/sommelier/v4/x/cork/types";
+
+message AddManagedCellarIDsProposal {
+  string title = 1;
+  string description = 2;
+  CellarIDSet cellar_ids = 3;
+}
+
+// AddManagedCellarIDsProposalWithDeposit is a specific definition for CLI commands
+message AddManagedCellarIDsProposalWithDeposit {
+  string title = 1;
+  string description = 2;
+  repeated string cellar_ids = 3;
+  string deposit = 4;
+}
+
+message RemoveManagedCellarIDsProposal {
+  string title = 1;
+  string description = 2;
+  CellarIDSet cellar_ids = 3;
+}
+
+// RemoveManagedCellarIDsProposalWithDeposit is a specific definition for CLI commands
+message RemoveManagedCellarIDsProposalWithDeposit {
+  string title = 1;
+  string description = 2;
+  repeated string cellar_ids = 3;
+  string deposit = 4;
+}
+
+message ScheduledCorkProposal {
+  string title = 1;
+  string description = 2;
+  uint64 block_height = 3;
+  string target_contract_address = 4;
+  string contract_call_proto_json = 5;
```

Comment:
Doing this made me realize we probably have to use a full `ScheduleRequest` proto in this field rather than just a function call. outlined the reasoning in chat but for context:

This is what I am currently using in my dev branch, but Steward would be unable to determine what Cellar API this is in reference to if there were two with two message that had the same name/structure:

```json
{
  “trust_position”: {
    “erc20_address: “…”
  }
}
```

It's annoying because these extra fields are already in the prop, but we probably just need to make the submitter use a who `ScheduleRequest` struct to ensure Steward can decode it to the right cellar type. The code I have so far just guesses based on whether something errors, but as mentioned above it could get a false positive if we do that.

```
// whole request
{
  “cellar_id”: “…”,
  “cellar_v1”: {
    “trust_position”: {
      “erc20_address: “…”
    }
  },
  block_height: 100000
}
```

I guess we can just have Steward validate that the prop's field and json field values match.

Reply:
Given this and for the simplicity of decoding, we should probably then use a fully qualified JSON blob including the type name to do this. Bear in mind the governance proto will be a separate entity from the Submit proto, so while it will have to mirror some of its structure it can skip the extraneous information.